src/installer/records.py

```diff
@@ -8,17 +9,53 @@
 import csv
 import warnings
 
-from installer._compat import pathlib
 from installer._compat.typing import TYPE_CHECKING
 
 if TYPE_CHECKING:
-    from typing import Iterator, Optional
+    from typing import Iterable, Iterator, Optional
 
 
 class SuperfulousRecordColumnsWarning(UserWarning):
     pass
 
 
+class Path(object):
+    """A light-weight path interface.
+
+    This is used like ``pathlib.PurePosixPath``, but slim.
+    """
+
+    __slots__ = ("_parts",)
+
+    def __init__(self, *parts):
+        # type: (*str) -> None
+        self._parts = tuple(self._iter_parts(parts))
+
+    @staticmethod
+    def _iter_parts(parts):
+        # type: (Iterable[str]) -> Iterator[str]
+        for part in parts:
+            for p in part.split("/"):
```

Comment:
still, not a fan of a single char variable if it's not inside a comprehension; and also find the iter keyword in there useful, especially with type annotations.

Reply:
Not even for a variable that’s only used in the immediately two lines after? IMHO this is a bit obessesive, but I’m OK if you feel strongly. What would be a good alternative? I can think of something like `split_part`, but that doesn’t read more easily to me.

Speaking of obssesive, I do feel very strongly about the `iter` part. The type hint is not useful on the call site, where the information is important. And it is very possible to write code that escape type checks but entirely not make sense:

```python
parts = self._split_parts(parts)  # This returns an iterator.
for part in parts:
   # ...
for part in parts:  # This is wrong.
   # ...
```

Sticking `iter` in the name is the best way IMO to avoid this class of coding errors.